src/api/AuthApi/AuthApi.ts

```diff
@@ -0,0 +1,36 @@
+import { AuthApiSignInRequest, AuthApiSignUpRequest, AuthApiSignUpResponse } from './AuthApi.types';
+
+const API_URL = 'https://ya-praktikum.tech/api/v2/auth';
+
+export const AuthApi = {
+  async signUp(data: AuthApiSignUpRequest): Promise<AuthApiSignUpResponse> {
+    const response = await fetch(`${API_URL}/signup`, {
```

Comment:
❗️
Если не хочется юзать `axios` или типа того, нужно обособить базовые http-методы самостоятельно. Тут видно, что для `signIn` и `signUp` всё , кроме урла, дублируется. И так будет и дальше с другими апишками. Общие части нужно вынести в отдельную api-обёртку, возможно выставив наружу для удобства готовые методы `post`/`get`/`etc`, чтобы тут можно было сделать, например:
```suggestion
    const response = ApiClient.post(url, { body })
```